### PR TITLE
Permit styles [Modal/Stepper/IconSpinner]

### PR DESCRIPTION
**Pull request by fairlighteth**
# Summary

Adds updated `Modal` `Stepper` and `IconSpinner` modules. To be used with the Permit flow. Available on Cosmos.

https://github.com/cowprotocol/cowswap/assets/31534717/da9af92b-fc9f-46fb-a812-9ca2d7e7404f

Screenshot 2023-08-31 at 15 06 35

https://github.com/cowprotocol/cowswap/assets/31534717/70acb256-8a1a-4888-99fa-427fcaa55c22

https://github.com/cowprotocol/cowswap/assets/31534717/54a39350-aff8-439c-8708-e3dd54df7463

### CODE REVIEW

File: apps/cowswap-frontend/src/common/pure/IconSpinner/index.tsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    content = <CurrencyLogo currency={currency} size="100%" />;
+  } else if (image) {
+    content = <img src={image} alt="Spinning icon" width={size} height={size} />;
+  } else if (children) {
+    content = <span>{children}</span>;
+  } else {
+    content = <span />;
+  }
+
+  return (
```
Comment (shoom3301) [inline apps/cowswap-frontend/src/common/pure/IconSpinner/index.tsx:80]: Just an idea how to write it shorter:

```
<Wrapper size={size} spinnerWidth={spinnerWidth} bgColor={bgColor}>
      {() => {
        if (currency) {
          return <CurrencyLogo currency={currency} size="100%" />
        } else if (image) {
          return <img src={image} alt="Spinning icon" width={size} height={size} />
        } else if (children) {
          return <span>{children}</span>
        }

        return <span />
      }}
    </Wrapper>
```

File: apps/cowswap-frontend/src/common/pure/Stepper/index.tsx [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+interface StepProps {
+  stepState: StepState
+  stepNumber: number
+  label: string
+  dotSize?: number
+  dotBorderColor?: string
+}
+
+const stateStyles = {
```
Comment (shoom3301) [inline apps/cowswap-frontend/src/common/pure/Stepper/index.tsx:17]: Please, specify a type for the const:
```
interface StepStyles {
  dotColor: string
  dotBorderColor: string
  labelColor: string
}

const stateStyles: Record<StepState, StepStyles> = {
```

**Review by alfetopito — APPROVED** [review]

File: apps/cowswap-frontend/src/common/pure/IconSpinner/index.tsx (hunk 1 above)
Comment (fairlighteth) [reply apps/cowswap-frontend/src/common/pure/IconSpinner/index.tsx:69]: Thank you for the suggestion, updated!

**Review by shoom3301 — APPROVED** [review]

**Comment by alfetopito** [comment]
Something is wrong with the types, the build is failing

image

File: apps/cowswap-frontend/src/common/pure/Modal/index.cosmos.tsx [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+`
+
+const ArrowRight = styled(SVG)`
+  --size: 12px;
+  width: var(--size);
+  height: var(--size);
+  margin: auto;
+
+  > path {
+    fill: var(--cow-color-text2);
```
Comment (alfetopito) [inline apps/cowswap-frontend/src/common/pure/Modal/index.cosmos.tsx:23]: Here the enum would be used:


```suggestion
    fill: var(${ColorTypes.COLOR_TEXT_2});
```

File: apps/cowswap-frontend/src/common/pure/Modal/index.tsx [hunk 4]
```diff
@@ -141,3 +145,169 @@ export const CowModal = styled(Modal)<{
...
+// New Modal to be used going forward =================================
+const ModalInner = styled.div`
+  display: flex;
+  flex-direction: column;
+  width: 100%;
+  height: auto;
+  margin: auto;
+  background: var(--cow-container-bg-01);
+  border-radius: var(--cow-border-radius-normal);
+  box-shadow: var(--cow-box-shadow-normal);
```
Comment (alfetopito) [inline apps/cowswap-frontend/src/common/pure/Modal/index.tsx:158]: And here

```suggestion
  background: var(${ColorTypes.CONTAINER_BG_01});
  border-radius: var(${ColorTypes.BORDER_RADIUS_NORMAL});
  box-shadow: var(${ColorTypes.BOX_SHADOW_NORMAL});
```

And so on

File: apps/cowswap-frontend/src/legacy/theme/baseTheme.tsx [hunk 5]
```diff
@@ -400,6 +400,18 @@ export const UniThemedGlobalStyle = css`
     --cow-color-yellow-light: ${({ theme }) => theme.alert2};
     --cow-color-green: ${({ theme }) => theme.success};
 
+    // Base
+    --cow-color-border: var(--cow-color-grey);
+    --cow-container-bg-01: ${({ theme }) => theme.bg1};
+    --cow-modal-backdrop: var(--cow-color-text1-opacity-25);
+    --cow-border-radius-normal: 24px;
+    --cow-padding-normal: 24px;
+    --cow-box-shadow-normal: 0 1.5rem 1rem #00000008,0 .75rem .75rem #0000000a,0 .25rem .375rem #0000000d;
```
Comment (alfetopito) [inline apps/cowswap-frontend/src/legacy/theme/baseTheme.tsx:409]: Here is a base place for their definition, using the same way.

It's just a template string, so it shouldn't add any extra processing/make it slower

```suggestion
    ${ColorTypes.COLOR_BORDER}: var(${ColorTypes.COLOR_GREY});
    --cow-container-bg-01: ${({ theme }) => theme.bg1};
    --cow-modal-backdrop: var(--cow-color-text1-opacity-25);
    --cow-border-radius-normal: 24px;
    --cow-padding-normal: 24px;
    --cow-box-shadow-normal: 0 1.5rem 1rem #00000008,0 .75rem .75rem #0000000a,0 .25rem .375rem #0000000d;
```

**Review by alfetopito — COMMENTED** [review]
Leaving some example of usage of the enum for variable names.

**Comment by fairlighteth** [comment]
Merging as I'm addressing the ENUMs in https://github.com/cowprotocol/cowswap/pull/3110

**Pull request merged**